### PR TITLE
support removal of keystore file if password incorrect

### PR DESCRIPTION
**Pull request by jessereynolds**
If the new `password_fail_reset` attribute is set to true, and if the supplied password does not succeed at unlocking the keystore file, then the keystore file will be removed and re-created. 

Default behaviour is unchanged.

### CODE REVIEW

**Comment by tphoney** [comment]
Hi @jessereynolds , thanks for the work you have put into this pull request. However due to inactivity and failing unit tests / needing rebased this pull request is being closed. If you want to re-submit this PR with the necessary changes, we will gladly look at the new pull request.

File: spec/acceptance/keystore_failed_password_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        password     => 'coraline',
+        trustcacerts => true,
+        path         => #{resource_path},
+    }
+    EOS
+    apply_manifest(pp, :catch_failures => true)
+  end
+
+  it 'verifies the keystore' do
+    shell("#{keytool_path}keytool -list -v -keystore /etc/keystore.ts -storepass coraline") do |r|
```
Comment (hunner) [inline spec/acceptance/keystore_failed_password_spec.rb]: Should this (and the other 3 occurances) be `-keystore /etc/keystore_failed_password.ts` to fix the failing acceptance tests?
```
keytool error: java.lang.Exception: Keystore file does not exist: /etc/truststore.ts
java.lang.Exception: Keystore file does not exist: /etc/truststore.ts
```

Comment (jessereynolds) [inline spec/acceptance/keystore_failed_password_spec.rb]: Yes!

**Comment by jessereynolds** [comment]
@hunner good to merge I think?

**Comment by hunner** [comment]
Thanks!

**Pull request merged**